apps/web/components/dashboard/BulkBookmarksAction.tsx

```diff
@@ -0,0 +1,204 @@
+import bulkActions from "@/store/bulkBookmarksAction";
+import { Archive, ArchiveRestore, Pencil, Star, Trash2, X } from "lucide-react";
+
+import {
+  useDeleteBookmark,
+  useUpdateBookmark,
+} from "@hoarder/shared-react/hooks/bookmarks";
+
+function DeleteModal({
+  open,
+  setOpen,
+  deleteBookmarks,
+}: {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  deleteBookmarks: () => void;
+}) {
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Delete Bookmarks</DialogTitle>
+          <DialogClose />
+        </DialogHeader>
+        <p>Are you sure you want to delete these bookmarks?</p>
+        <DialogFooter>
+          <Button variant="ghost" onClick={() => setOpen(false)}>
+            Cancel
+          </Button>
+          <Button
+            variant="destructive"
+            onClick={() => {
+              deleteBookmarks();
+              setOpen(false);
+            }}
+          >
+            Delete
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
+  );
+}
+
+export default function BulkBookmarksAction() {
+  const { selectedBookmarks, isBulkEditEnabled } = bulkActions();
+  const handleBulkEdit = bulkActions((state) => state.handleBulkEdit);
+  const { toast } = useToast();
+  const pathname = usePathname();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+
+  useEffect(() => {
+    handleBulkEdit(false); // turn off toggle + clear selected bookmarks on mount
+  }, []);
+
+  const onError = () => {
+    toast({
+      variant: "destructive",
+      title: "Something went wrong",
+      description: "There was a problem with your request.",
+    });
+  };
+
+  const deleteBookmarkMutator = useDeleteBookmark({
+    onSuccess: () => {
+      handleBulkEdit(false);
+    },
+    onError,
+  });
+
+  const updateBookmarkMutator = useUpdateBookmark({
+    onSuccess: () => {
+      handleBulkEdit(false);
+    },
+    onError,
+  });
+
+  interface UpdateBookmarkProps {
+    favourited?: boolean;
+    archived?: boolean;
+  }
+
+  const updateBookmarks = ({
+    favourited,
+    archived,
+  }: UpdateBookmarkProps): void => {
+    selectedBookmarks.map((item) => {
+      updateBookmarkMutator.mutate({
+        bookmarkId: item.id,
+        favourited: favourited ?? item.favourited,
+        archived: archived ?? item.archived,
+      });
+    });
+    toast({
+      description: `${selectedBookmarks.length} bookmarks have been updated!`,
+    });
+  };
+
+  const deleteBookmarks = () => {
+    selectedBookmarks.map((item) => {
+      deleteBookmarkMutator.mutate({ bookmarkId: item.id });
+    });
+    toast({
+      description: `${selectedBookmarks.length} bookmarks have been deleted!`,
+    });
+  };
+
+  const alreadyFavourited =
+    (selectedBookmarks.length &&
+      selectedBookmarks.every((item) => item.favourited === true)) ||
+    pathname.includes("favourites");
```

Comment:
Is this check really needed? Isn't it kinda redundant? I don't like the path matching here tbh, kinda fragile.

Reply:
- I added this check if you're on a list and you select multiple items that are already favourite/archived, you can un-favourite/un-archive at once. Good to have. 
- Agree with path checking. What do you suggest? How should we handle the case where we want to show 'un-favorite' and 'un-archive' icons on the Favorites and Archive pages?

Reply:
If the selected bookmark are in favorites wouldn't that match the condition before the path check anyways?